xwiki-platform-core/xwiki-platform-mail/xwiki-platform-mail-send/xwiki-platform-mail-send-default/src/main/java/org/xwiki/mail/script/ScriptMimeMessage.java

```diff
@@ -101,28 +114,34 @@ public BodyPart addPart(String mimeType, Object content)
      * @return the Mime Body Part object that was added. Returning it allows script to make modifications to that body
      *         part after it's been set (get/set some headers, etc)
      * @param mimeType the mime type of the content parameter
-     * @param content the content to include in the mail. The type depends on the mimetype. For example for a mime
-     *        type of {@code text/plain} or {@code text/html}, the content should be a String, for a mime type of
-     *        {@code xwiki/template} it should be a {@link org.xwiki.model.reference.DocumentReference}, etc.
+     * @param content the content to include in the mail. The type depends on the mimetype. For example for a mime type
+     *            of {@code text/plain} or {@code text/html}, the content should be a String, for a mime type of
+     *            {@code xwiki/template} it should be a {@link org.xwiki.model.reference.DocumentReference}, etc. Also
+     *            accepts a {@link BodyPart} object, in which case, the mime type and parameters are not used.
      * @param parameters the list of extra parameters. This is used for example to pass alternate content for the mail
-     *        using the {@code alternate} key in the HTML Mime Body Part Factory. Mail headers can also be passed using
-     *        the {@code headers} key with a {@code Map&lt;String, String&gt;} value containing header keys and values
+     *            using the {@code alternate} key in the HTML Mime Body Part Factory. Mail headers can also be passed
+     *            using the {@code headers} key with a {@code Map&lt;String, String&gt;} value containing header keys
+     *            and values.
      */
     public BodyPart addPart(String mimeType, Object content, Map<String, Object> parameters)
     {
         BodyPart bodyPart;
 
         try {
-            MimeBodyPartFactory factory = getBodyPartFactory(mimeType, content.getClass());
+            if (!(content instanceof BodyPart)) {
```

Comment:
I'm not sure I like this since it means accepting a null mimetype and a null parameters. This is not what this method was meant to be. So I'd remove all the changes you brought to this method.

Reply:
Hi @vmassol, I'm not sure what to do as I'm hearing different things from you and Thomas. He wants to add the functionality to this method to avoid running the extra code when the object is already a BodyPart and also to support calls from the other method while you think that it is unnecessary and it is not a good idea to accept null parameters.

Reply:
@akshitdewan You're perfectly right: @tmortagne and I need to agree first :) Sorry about that.